### PR TITLE
feat: added endpoint for generating short-form referral link

### PR DESCRIPTION
**Pull request by edustipend-root**
Added a new endpoint POST `/v1/referral/link` for generating referral links.

**Long-form URL**
 `https://www.edustipend.org/support-a-learner?utm_referrer=Jay Kay&utm_medium=jaykay@yahoo.co.uk&utm_campaign=Edustipend%20At%202&utm_source=instagram` 

**Shortened URL**
`https://go.edustipend.org/8uIl0W`

### CODE REVIEW

File: utils/urlEncoder.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+require("dotenv").config();
+
+const constructEncodedSALReferralUrl = (
+  name,
+  email,
+  campaign = "Edustipend at 2",
+  source
+) => {
+  const referralUrl = `https://www.edustipend.org/support-a-learner?utm_campaign=${campaign}&utm_medium=${email}&utm_referrer=${name}&utm_source=${
+    source ?? ""
```
Comment (Chiemezuo) [inline utils/urlEncoder.js:10]: Just out of curiosity, what are the different sources we can have? (Referral link systems are new to me)

File: services/Referral.js [hunk 2]
```diff
@@ -19,6 +22,84 @@ class Referral {
...
+        Logger.info(`Link already created for ${email}, returning entry in DB`);
+        return referralLink;
+      }
+    } catch (e) {
+      Logger.error(`Error finding referral link for ${email}`, e);
+    }
+
+    // Create long form encoded URL
+    try {
+      longFormEncodedUrl = constructEncodedSALReferralUrl(name, email);
```
Comment (Chiemezuo) [inline services/Referral.js:51]: What does "SAL" mean in `constructEncodedSALReferralUlr`

**Comment by Chiemezuo** [comment]
The changes look good to me.

**Review by Chiemezuo — APPROVED** [review]

File: utils/urlEncoder.js (hunk 1 above)
Comment (edustipend-root) [reply utils/urlEncoder.js:10]: Here: https://buffer.com/library/utm-guide/

File: services/Referral.js (hunk 2 above)
Comment (edustipend-root) [reply services/Referral.js:51]: `Support A Learner`

**Pull request merged**